core/src/test/java/de/danoeh/antennapod/core/storage/ItemEnqueuePositionCalculatorTest.java

```diff
@@ -0,0 +1,302 @@
+                            concat(list(101L, 102L, 201L), QUEUE_DEFAULT_IDS),
+                            concat(list(101L, 102L, 201L, 202L), QUEUE_DEFAULT_IDS),
+                            FRONT, QUEUE_DEFAULT},
+            });
+        }
+
+        @Parameter
+        public String message;
+
+        @Parameter(1)
+        public List<Long> idsExpectedAfter101;
+
+        @Parameter(2)
+        public List<Long> idsExpectedAfter102;
+
+        // 2XX are for testing bulk insertion cases
+        @Parameter(3)
+        public List<Long> idsExpectedAfter201;
+
+        @Parameter(4)
+        public List<Long> idsExpectedAfter202;
+
+        @Parameter(5)
+        public EnqueueLocation options;
+
+        @Parameter(6)
+        public List<FeedItem> queueInitial;
+
+        @Test
+        public void testQueueOrderWhenDownloading2Items() {
+
+            // Setup class under test
+            //
+            ItemEnqueuePositionCalculator calculator = new ItemEnqueuePositionCalculator(options);
+            DownloadStateProvider stubDownloadStateProvider = mock(DownloadStateProvider.class);
+            stub(stubDownloadStateProvider.isDownloadingFile(any(FeedMedia.class))).toReturn(false);
+            calculator.downloadStateProvider = stubDownloadStateProvider;
+
+            // Setup initial data
+            // A shallow copy, as the test code will manipulate the queue
+            List<FeedItem> queue = new ArrayList<>(queueInitial);
+
+
+            // Test body
+
+            // User clicks download on feed item 101
+            FeedItem tFI101 = tFI_isDownloading(101, stubDownloadStateProvider);
+            doAddToQueueAndAssertResult(message + " (1st download)",
+                    calculator, 0, tFI101, queue,
+                    idsExpectedAfter101);
+
+            // Then user clicks download on feed item 102
+            FeedItem tFI102 = tFI_isDownloading(102, stubDownloadStateProvider);
+            doAddToQueueAndAssertResult(message + " (2nd download, it should preserve order of download)",
+                    calculator, 0, tFI102, queue,
+                    idsExpectedAfter102);
+
+            // Items 201 and 202 are added as part of a single DBWriter.addQueueItem() calls
+
+            FeedItem tFI201 = tFI_isDownloading(201, stubDownloadStateProvider);
+            doAddToQueueAndAssertResult(message + " (bulk insertion, 1st item)",
+                    calculator, 0, tFI201, queue,
+                    idsExpectedAfter201);
+
+            FeedItem tFI202 = tFI_isDownloading(202, stubDownloadStateProvider);
+            doAddToQueueAndAssertResult(message + " (bulk insertion, 2nd item)",
+                    calculator, 1, tFI202, queue, null,
+                    idsExpectedAfter202);
+
+            // TODO: simulate download failure cases.
+        }
+
+
+        private static FeedItem tFI_isDownloading(int id, DownloadStateProvider stubDownloadStateProvider) {
```

Comment:
What is tFI? Also, please do not use underscores in method names.

Reply:
renamed to `setAsDownloading` . 

A clarification is needed for future reference. For not using underscore in method names, are you talking about this case specifically or in general? 

A common convention for test methods is that, different cases of the same test would be in the form of `aTestMethod_case1`? Are we not following this convention?
(I know the convention is not applicable to the above method. That's why I'm asking if your comment is meant to be a general one or just specific to this case).